File: scripts/src/tools/utils/sysinfo/plain.py
from tools.core.console import out
from tools.utils.sysinfo.branding import resolve_brand
from tools.utils.sysinfo.health import health_summary
from tools.utils.sysinfo.models import HealthIssue, RenderOptions, SystemView


def component_identity(component):
    brand = resolve_brand(component.kind, component.vendor, component.model, *component.identifiers)
    if brand.key == component.kind:
        return component.model
    return f"{brand.name} {component.model}".strip()


def render_health(issues):
    for issue in issues:
        out(f"{issue.severity.title()}: {issue.title}")
        if issue.detail:
            out(f"  {issue.detail}")
        if issue.action:
            out(f"  Action: {issue.action}")


def render_compact(view, issues):
    out("System: " + "  ".join(view.summary))
    for component in view.components:
        if component.compact:
            out(f"{component.label}: {component_identity(component)}")
    summary = health_summary(issues)
    if summary:
        out(f"Health: {summary}")


def render_full(view, issues):
    out("System")
    out(f"  Platform: {view.platform.label}")
    for item in view.system_facts:
        out(f"  {item.label}: {item.value}")
    out("Hardware")
    for component in view.components:
        out(f"  {component.label}: {component_identity(component)}")
        for item in component.facts:
            out(f"    {item.label}: {item.value}")
    if view.software:
        out("Software")
        for badge in view.software:
            out(f"  {badge.kind.title()}: {badge.label}")
    summary = health_summary(issues)
    if summary:
        out(f"Health: {summary}")


def render_plain(view: SystemView, issues: tuple[HealthIssue, ...], options: RenderOptions):
    if options.full:
        render_full(view, issues)
    else:
        render_compact(view, issues)
    if options.health and issues:
        out()
        render_health(issues)

File: scripts/rust/crates/sysinfo/src/presentation/plain.rs
use super::branding::resolve_brand;
use crate::health::health_summary;
use crate::model::{Component, HealthIssue, RenderOptions, SystemView};

pub fn component_identity(component: &Component) -> String {
    let mut identities = vec![component.vendor.as_str(), component.model.as_str()];
    identities.extend(component.identifiers.iter().map(String::as_str));
    let brand = resolve_brand(&component.kind, &identities);
    if brand.key == component.kind {
        component.model.clone()
    } else {
        format!("{} {}", brand.name, component.model)
            .trim()
            .to_string()
    }
}
pub fn render_plain(view: &SystemView, issues: &[HealthIssue], options: RenderOptions) -> String {
    let mut lines = Vec::new();
    if options.full {
        lines.push("System".into());
        lines.push(format!("  Platform: {}", view.platform.label));
        lines.extend(
            view.system_facts
                .iter()
                .map(|fact| format!("  {}: {}", fact.label, fact.value)),
        );
        lines.push("Hardware".into());
        for component in &view.components {
            lines.push(format!(
                "  {}: {}",
                component.label,
                component_identity(component)
            ));
            lines.extend(
                component
                    .facts
                    .iter()
                    .map(|fact| format!("    {}: {}", fact.label, fact.value)),
            );
        }
        if !view.software.is_empty() {
            lines.push("Software".into());
            for badge in &view.software {
                lines.push(format!("  {}: {}", titlecase(&badge.kind), badge.label));
            }
        }
    } else {
        lines.push(format!("System: {}", view.summary.join("  ")));
        lines.extend(
            view.components
                .iter()
                .filter(|component| component.compact)
                .map(|component| format!("{}: {}", component.label, component_identity(component))),
        );
    }
    let summary = health_summary(issues);
    if !summary.is_empty() {
        lines.push(format!("Health: {summary}"));
    }
    if options.health && !issues.is_empty() {
        lines.push(String::new());
        for issue in issues {
            lines.push(format!(
                "{}: {}",
                titlecase(issue.severity.as_str()),
                issue.title
            ));
            if !issue.detail.is_empty() {
                lines.push(format!("  {}", issue.detail));
            }
            if !issue.action.is_empty() {
                lines.push(format!("  Action: {}", issue.action));
            }
        }
    }
    lines.join("\n") + "\n"
}
fn titlecase(value: &str) -> String {
    let mut chars = value.chars();
    match chars.next() {
        Some(first) => first.to_uppercase().collect::<String>() + chars.as_str(),
        None => String::new(),
    }
}
